live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java

```diff
@@ -54,39 +63,113 @@ public class RepoEventMapper
 
     public IngestContentCommand mapToIngestContentCommand(RepoEvent<DataAttributes<NodeResource>> event)
     {
+        ensureThat(isEventTypeCreated(event), "Unsupported event type");
+
         return new IngestContentCommand(
                 event.getTime().toInstant().toEpochMilli(),
                 event.getData().getResource().getId());
     }
 
     public IngestMetadataCommand mapToIngestMetadataCommand(RepoEvent<DataAttributes<NodeResource>> event)
     {
-        ensureThat(isEventTypeCreated(event), "Unsupported event type");
+        ensureThat(isEventTypeCreated(event) || isEventTypeUpdated(event), "Unsupported event type");
 
         return new IngestMetadataCommand(
-                event.getTime().toInstant().toEpochMilli(),
+                toMilliseconds(event.getTime()),
                 event.getData().getResource().getId(),
-                updated(event.getData().getResource().getName()),
-                updated(event.getData().getResource().getPrimaryAssocQName()),
-                updated(event.getData().getResource().getNodeType()),
-                updated(event.getData().getResource().getCreatedByUser().getId()),
-                updated(event.getData().getResource().getModifiedByUser().getId()),
-                updated(event.getData().getResource().getAspectNames()),
-                updated(event.getData().getResource().isFile()),
-                updated(event.getData().getResource().isFolder()),
-                updated(event.getData().getResource().getCreatedAt().toInstant().toEpochMilli()),
-                allCustomPropertiesUpdated(event));
+                calculatePropertyDelta(event, NodeResource::getName),
```

Comment:
If `node` is nullable it won't fail on NPE, while trying to obtain the `name` and rest of the data?

Reply:
`nodeResourceBefore` is nullable but later on we are wrapping it with an optional to ensure it won't fail:
<img width="953" alt="image" src="https://github.com/Alfresco/hxi-connector/assets/62990104/755a8cd4-5d6a-42ba-8fc7-3f158f9c37df">